### PR TITLE
🌱 Bump vmoperator to v1.8.5 & vmware dependencies

### PR DESCRIPTION
**Pull request by sbueringer**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -15,10 +15,10 @@ require (
...
 	github.com/stretchr/testify v1.8.4
-	github.com/vmware-tanzu/net-operator-api v0.0.0-20210401185409-b0dc6c297707
-	github.com/vmware-tanzu/vm-operator/api v1.8.3-0.20231114230806-852c1641447a
-	github.com/vmware-tanzu/vm-operator/external/ncp v0.0.0-20211209213435-0f4ab286f64f
-	github.com/vmware-tanzu/vm-operator/external/tanzu-topology v0.0.0-20211209213435-0f4ab286f64f
+	github.com/vmware-tanzu/net-operator-api v0.0.0-20231019160108-42131d6e8360
+	github.com/vmware-tanzu/vm-operator/api v1.8.5
+	github.com/vmware-tanzu/vm-operator/external/ncp v0.0.0-20231214185006-5477585eebfd
+	github.com/vmware-tanzu/vm-operator/external/tanzu-topology v0.0.0-20231214185006-5477585eebfd
 	github.com/vmware/govmomi v0.34.0
```
Comment (sbueringer) [inline go.mod:22]: Checked:
* github.com/vmware-tanzu/net-operator-api
* github.com/vmware-tanzu/vm-operator/external/ncp
* github.com/vmware-tanzu/vm-operator/external/tanzu-topology

Each of these only contain one CRD that we use. The CRDs either didn't change or we are not reading any fields (AvailabilityZone)


Based on Sagar's guidance it should be fine to bump vm-operator to v1.8.5

**Comment by sbueringer** [comment]
/hold

Let's merge the test module thing first

**Comment by sbueringer** [comment]
Rebased on top of #2532. Will drop the underlying commit once #2532 is merged

**Review by chrischdi — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**